lib/improver/tests/psychrometric_calculations/precip_phase_probability/test_ProbPhase.py

```diff
@@ -0,0 +1,176 @@
+        }
+
+        data = np.zeros((3, 3), dtype=np.float32)
+
+        orog_cube = set_up_variable_cube(
+            data, name='surface_altitude', units='m',
+            spatial_grid='equalarea', attributes=self.mandatory_attributes)
+
+        falling_level_data = np.array(
+            [[[-1, -1, -1],
+              [-1, -1, -1],
+              [-1, -1, -1]],
+             [[0, -1, 0],
+              [0, 1, 0],
+              [0, -1, 0]],
+             [[1, 1, 1],
+              [1, 1, 1],
+              [1, 1, 1]]], dtype=np.float32)
+
+        falling_level_cube = set_up_variable_cube(
+            falling_level_data, units='m', spatial_grid='equalarea',
+            name='altitude_of_snow_falling_level', realizations=[0, 1, 2],
+            attributes=self.mandatory_attributes)
+
+        self.cubes = iris.cube.CubeList([falling_level_cube, orog_cube])
+
+    def test_prob_snow(self):
+        """Test that process returns a cube with the right name, units and
+        values. In this instance the phase change is from snow to sleet."""
+        result = self.plugin.process(self.cubes)
+        expected = np.zeros((3, 3, 3), dtype=np.float32)
+        expected[0] = 1.
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.name(), "probability_of_snow_at_surface")
+        self.assertEqual(result.units, Unit('1'))
+        self.assertDictEqual(result.attributes, self.mandatory_attributes)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_prob_rain(self):
+        """Test that process returns a cube with the right name, units and
+        values. In this instance the phase change is from sleet to rain."""
+        self.cubes[1].units = Unit('feet')
+        self.cubes[0].rename('altitude_of_rain_falling_level')
+        result = self.plugin.process(self.cubes)
+        expected = np.zeros((3, 3, 3), dtype=np.float32)
+        expected[2] = 1.
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.name(), "probability_of_rain_at_surface")
+        self.assertEqual(result.units, Unit('1'))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_unit_conversion(self):
+        """Test that process returns the same as test_prob_rain when the
+        orography cube units are in feet."""
+        self.cubes[0].rename('altitude_of_rain_falling_level')
+        result = self.plugin.process(self.cubes)
+        expected = np.zeros((3, 3, 3), dtype=np.float32)
+        expected[2] = 1.
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.name(), "probability_of_rain_at_surface")
+        self.assertEqual(result.units, Unit('1'))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_bad_phase_cube(self):
+        """Test that process raises an exception when the input phase cube is
+        incorrectly named."""
+        self.cubes[0].rename('altitude_of_kittens')
```

Comment:
Good use of kittens. I'm envisaging something like a tree line above which you don't find any kittens.